nemo_curator/modules/semantic_dedup/clusteringmodel.py

```diff
@@ -56,10 +62,10 @@ def __init__(
         clustering_output_dir: str = "./clustering_results",
         embedding_column: str = "embeddings",
         random_state: int = 1234,
-        sim_metric: str = "cosine",
-        which_to_keep: str = "hard",
+        # TODO : add l2 distance support
+        sim_metric: Literal["cosine"] = "cosine",
```

Comment:
Is it possible to start creating code blocks for where the differences between `sim_metric="cosine"` and `sim_metric="l2"`/`sim_metric="euclidean"` will be? Maybe with a `NotImplementedError` for now? Just to give me a more general idea of how it will look.

Also, I was wondering if there would be any benefit to allowing the user to pass in their own `sim_metric` function here if they want to?

Reply:
Also asking because cosine is a similarity metric while L2 is a dissimilarity metric, so will some of the logic be reversed?

Reply:
At this moment it'll be hard to support an arbitrary metric, but something we can definitely add. In another project that uses SemDedup, they have a use case of preferring rows that have a certain higher value in some metadata, in which case they might want to perform a join or something like that.